pkg/scheduler/api/compatibility/compatibility_test.go

```diff
@@ -24,29 +24,77 @@ import (
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes/fake"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
 	_ "k8s.io/kubernetes/pkg/apis/core/install"
+	"k8s.io/kubernetes/pkg/features"
 	"k8s.io/kubernetes/pkg/scheduler"
+	"k8s.io/kubernetes/pkg/scheduler/algorithmprovider"
 	_ "k8s.io/kubernetes/pkg/scheduler/algorithmprovider/defaults"
 	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
 	kubeschedulerconfig "k8s.io/kubernetes/pkg/scheduler/apis/config"
 	schedulerconfig "k8s.io/kubernetes/pkg/scheduler/apis/config"
 	"k8s.io/kubernetes/pkg/scheduler/core"
 )
 
-func TestCompatibility_v1_Scheduler(t *testing.T) {
-	snapshot := scheduler.RegisteredPredicatesAndPrioritiesSnapshot()
-	defer scheduler.ApplyPredicatesAndPriorities(snapshot)
```

Comment:
A more straightforward solution is to add a parameter with type `func()` to the original "schedulerFiles" - which can be renamed to "testcases".
And also the struct `map[string]struct` can be refactored to `[]struct`, and add a field called "name" into the anonymous struct - which then follows the Golang subtest convention.

In terms of the new `func()` field, the subtest can choose to implement as a no-op or apply specific alpha/beta featuregate, and inside the main test body (the `for ... range testcases` loop), we call the `func()` in the beginning, and then call `defer algorithmprovider.ApplyFeatureGates()()`.

Reply:
Thanks @Huang-Wei. Yes, this will make the tests to follow the Golang subtest convention. My previous solution intends to apply the alpha/beta featuregate to all of the tests from 1.0 to 1.16, so this is not needed and we just need to add two alpha/beta featuregate examples based on 1.16, right?